browser/src/slideshow/SlideShowPresenter.ts

```diff
@@ -152,9 +152,11 @@ class SlideShowPresenter {
 		window.addEventListener('keydown', this._onCanvasKeyDown.bind(this));
 
 		try {
+			this._slideRenderer = new SlideRendererGl();
 			this._slideRenderer.setup(canvas);
 		} catch (error) {
-			this._slideRenderer.setup2d(canvas);
+			this._slideRenderer = new SlideRenderer2d();
+			this._slideRenderer.setup(canvas);
```

Comment:
Is there any reason to create a setup method? If not, then we can accept the canvas as a constructor parameter so we don't need to call the setup method every time.

Reply:
good point